src/azure-cli/azure/cli/command_modules/synapse/_params.py

```diff
@@ -251,29 +251,44 @@ def load_arguments(self, _):
-                   help='Indicates whether using the secondary storeage key or not')
-        c.argument('is_azure_monitor_target_enabled', options_list=['--enable-azure-monitor'],
-                   help='Whether enabling azure monitor target or not.',
-                   arg_type=get_three_state_flag())
-        c.argument('state',
-                   arg_group=policy_arg_group,
-                   help='Auditing policy state',
-                   arg_type=get_enum_type(BlobAuditingPolicyState))
-        c.argument('audit_actions_and_groups',
-                   options_list=['--actions'],
-                   arg_group=policy_arg_group,
-                   help='List of actions and action groups to audit.',
-                   nargs='+')
-        c.argument('retention_days',
-                   arg_group=policy_arg_group,
-                   help='The number of days to retain audit logs.')
+    for scope in ['synapse sql pool audit-policy', 'synapse sql audit-policy']:
+        with self.argument_context(scope + ' update') as c:
+            _configure_security_or_audit_policy_storage_params(c)
+            c.argument('storage_account_subscription_id', arg_group=storage_arg_group,
+                       options_list=['--storage-subscription'],
+                       help='The subscription id of storage account')
+            c.argument('is_storage_secondary_key_in_use', arg_group=storage_arg_group,
+                       arg_type=get_three_state_flag(), options_list=['--use-secondary-key'],
+                       help='Indicates whether using the secondary storeage key or not')
+            c.argument('is_azure_monitor_target_enabled', options_list=['--enable-azure-monitor'],
+                       help='Whether enabling azure monitor target or not.',
+                       arg_type=get_three_state_flag())
+            c.argument('state',
+                       arg_group=policy_arg_group,
+                       help='Auditing policy state',
+                       arg_type=get_enum_type(BlobAuditingPolicyState))
+            c.argument('audit_actions_and_groups',
+                       options_list=['--actions'],
+                       arg_group=policy_arg_group,
+                       help='List of actions and action groups to audit.',
+                       nargs='+')
+            c.argument('retention_days',
+                       arg_group=policy_arg_group,
+                       help='The number of days to retain audit logs.')
+
+    with self.argument_context('synapse sql audit-policy update') as c:
+        c.argument('queue_delay_milliseconds', type=int,
+                   options_list=['--queue-delay-time', '--queue-delay-milliseconds'],
+                   help='The amount of time in milliseconds that can elapse before audit actions are forced to be processed')
+
+    with self.argument_context('synapse sql ad-admin') as c:
+        c.argument('workspace_name', help='The workspace name.')
```

Comment:
if workspace is the target to operate, recommend use `--name` or `-n`.

Reply:
> if workspace is the target to operate, recommend use `--name` or `-n`.

I think it may be confused if we use `--name`. Because the cmdlet is `az synapse **sql** ad-admin`.